.github/workflows/ci_workflows.yml

```diff
@@ -110,3 +108,34 @@ jobs:
         - name: (Allowed Failure) Python 3.12 with remote data and dev version of key dependencies
           linux: py312-test-devdeps
           posargs: --remote-data=any --verbose
+
+  report_diff_test_coverage:
+    needs: [ tests ]
+    if: always() && github.event_name == 'pull_request'
+    name: report diff test coverage
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/setup-python@v5
+        with:
+          python-version: 3
+      - run: pip install diff-cover coverage
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - run: pip install .
+      - uses: actions/download-artifact@v4
+        with:
+          pattern: ".coverage*"
+          merge-multiple: true
+      - run: |
+          [ "$(ls -A .coverage*)" ] && exit 0 || exit 1
+      - run: coverage xml
+      - run: diff-cover coverage.xml --compare-branch origin/${{ github.base_ref }} --markdown-report diff_coverage.md
+      - run: cat diff_coverage.md >> $GITHUB_STEP_SUMMARY
```

Comment:
Why can't all 4 commands be in the same step?

Reply:
They could, but if one of them fails it'll be unclear at first glance what went wrong. IMO it's better for debugging and clarity to have them in separate steps